main.go

```diff
@@ -134,12 +142,31 @@ func main() {
 
 		http.HandleFunc("/status", h.StatusPageHandler)
 		http.HandleFunc("/", maybeAuth(h.CacheHandler, c.HtpasswdFile, c.Host))
+		httpServer := &http.Server{Addr: c.Host+":"+strconv.Itoa(c.Port), Handler: nil}
+
+		if c.IdleTimeout > 0 {
+			cache.LastRequestTime = time.Now()
+			ticker := time.NewTicker(time.Second)
+			go func() {
+				for {
+					select {
+					case <-ticker.C:
+						currentTime := time.Now()
+						lastRequestTime := cache.LastRequestTime
```

Comment:
I think you need either a Mutex or a Channel here since this var is being accessed from multiple threads?

Reply:
absolutely!